subprojects/android-test/ui-testing-core-app/src/androidTest/kotlin/com/avito/android/ui/test/ActivityLifecycleStatesTest.kt

```diff
@@ -0,0 +1,52 @@
+package com.avito.android.ui.test
+
+import androidx.lifecycle.Lifecycle
+import com.avito.android.test.app.core.inHouseScreenRule
+import com.avito.android.ui.EmptyActivity
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import ru.avito.util.assertThrows
+
+class ActivityLifecycleStatesTest {
+
+    @get:Rule
+    val activityRule = inHouseScreenRule<EmptyActivity>()
+
+    @Before
+    fun before() {
+        activityRule.launchActivity(null)
+    }
+
+    @Test
+    fun finishedActivity_isInDestroyedState_succeed() {
+        activityRule.scenario.onActivity { it.finish() }
+        activityRule.checks.isInState(Lifecycle.State.DESTROYED)
+    }
+
+    @Test
+    fun finishedActivity_isNotInDestroyedState_throwsAssertionError() {
+        activityRule.scenario.onActivity { it.finish() }
+        assertThrows<AssertionError> { activityRule.checks.isNotInState(Lifecycle.State.DESTROYED) }
+    }
+
+    @Test
+    fun launchedActivity_isNotInDestroyedState_succeed() {
```

Comment:
`isNotInDestroyedState` --> `isNotDestroyed` ?
It looks enough to express the idea.